src/backend/marsha/core/api/video.py

```diff
@@ -465,6 +478,110 @@ def update_live_state(
 
         return Response({"success": True})
 
+    @action(
+        methods=["get"],
+        detail=True,
+        url_path="pairing-secret",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def pairing_secret(
+        self,
+        request,
+        pk=None,
+    ):
+        """Generate a secret for pairing an external device to a live stream.
+
+        Deletes expired LivePairing objects.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the generated secret.
+        """
+        LivePairing.delete_expired()
+
+        video = self.get_object()
+        if video.live_type != JITSI:
+            return Response(
+                {"detail": "Matching video is not a Jitsi Live."}, status=400
+            )
+
+        try:
+            live_pairing = LivePairing.objects.get(video=video)
+        except LivePairing.DoesNotExist:
+            live_pairing = LivePairing(video=video)
+        live_pairing.generate_secret()
+        live_pairing.save()
```

Comment:
```suggestion
        for _ in range(2):
            try:
                live_pairing.generate_secret()
                live_pairing.save()
                break
            except ValidationError:
                pass
```
Let's try a compromise @sampaccoud :
Here we're keeping secret generation as is, but we're adding a second attempt to save it at the API level.
WDYT ?

Reply:
ok I will approve this version but I still find that the `generate_secret` method is complicated for nothing when a simple try/except does the job :wink: